### DISCUSSION TITLE
AIR SDK 33 or later Bug? Failed to read the external file on macOS

### DISCUSSION THREAD

**Discussion by zhstar** [airsdk/Adobe-Runtime-Support#2518] [Q&A]
Published as an app, the same code can work successfully in version 32, but failed in version 33 and later.

`
```
import flash.filesystem.*;

var file:File = File.desktopDirectory.resolvePath("1.txt");
var fileStream:FileStream = new FileStream();
fileStream.addEventListener(Event.COMPLETE, completeHandler)
fileStream.openAsync(file, FileMode.READ);
var results:ByteArray = new ByteArray();

function completeHandler(e:Event):void {
	
	if (fileStream.bytesAvailable > 0) {
		fileStream.readBytes(results);
		results.position = 0;
		var str:String = results.readMultiByte(results.length, "utf-8");
		txt.text = str;
		trace(str)
	}else {
		txt.text = ""
	}
}
```
`

**Comment by ajwfrost** [comment]
Hi

This code works for us .. can you please confirm which macOS version you're using, plus how you're actually packaging/running this? It sounds like you've published it as an application, have you code-signed it and does it have permissions to access the Desktop folder?

thanks

**Reply by zhstar** [reply]
Hi, I released the app directly using animate. How do I set the permission of the app.
IDE: Adobe Animate 2021
macOS: 11.6.8
